### DISCUSSION TITLE
cpp_infer demo build failed ,

### DISCUSSION THREAD

**Discussion by haishi11** [PaddlePaddle/PaddleOCR#12235] [Q&A]
- Build files have been written to: /home/cps/PaddleOCR/deploy/cpp_infer/build33
[ 14%] Building CXX object CMakeFiles/ppocr.dir/src/args.cpp.o
[ 14%] Building CXX object CMakeFiles/ppocr.dir/src/clipper.cpp.o
[ 28%] Building CXX object CMakeFiles/ppocr.dir/src/ocr_cls.cpp.o
[ 28%] Building CXX object CMakeFiles/ppocr.dir/src/ocr_det.cpp.o
[ 50%] Building CXX object CMakeFiles/ppocr.dir/src/postprocess_op.cpp.o
[ 50%] Building CXX object CMakeFiles/ppocr.dir/src/main.cpp.o
[ 50%] Building CXX object CMakeFiles/ppocr.dir/src/paddlestructure.cpp.o
[ 57%] Building CXX object CMakeFiles/ppocr.dir/src/preprocess_op.cpp.o
[ 85%] Building CXX object CMakeFiles/ppocr.dir/src/structure_table.cpp.o
[ 85%] Building CXX object CMakeFiles/ppocr.dir/src/ocr_rec.cpp.o
[ 85%] Building CXX object CMakeFiles/ppocr.dir/src/structure_layout.cpp.o
[ 85%] Building CXX object CMakeFiles/ppocr.dir/src/utility.cpp.o
[ 92%] Building CXX object CMakeFiles/ppocr.dir/src/paddleocr.cpp.o
/home/cps/PaddleOCR/deploy/cpp_infer/src/ocr_rec.cpp: 在成员函数‘void PaddleOCR::CRNNRecognizer::LoadModel(const string&)’中:
/home/cps/PaddleOCR/deploy/cpp_infer/src/ocr_rec.cpp:150:16: 错误：‘using Config = struct paddle::AnalysisConfig {aka struct paddle::AnalysisConfig}’ has no member named ‘CollectShapeRangeInfo’
         config.CollectShapeRangeInfo("./trt_rec_shape.txt");
                ^
/home/cps/PaddleOCR/deploy/cpp_infer/src/ocr_rec.cpp:152:16: 错误：‘using Config = struct paddle::AnalysisConfig {aka struct paddle::AnalysisConfig}’ has no member named ‘EnableTunedTensorRtDynamicShape’
         config.EnableTunedTensorRtDynamicShape("./trt_rec_shape.txt", true);
                ^
/home/cps/PaddleOCR/deploy/cpp_infer/src/ocr_cls.cpp: 在成员函数‘void PaddleOCR::Classifier::LoadModel(const string&)’中:
/home/cps/PaddleOCR/deploy/cpp_infer/src/ocr_cls.cpp:121:16: 错误：‘using Config = struct paddle::AnalysisConfig {aka struct paddle::AnalysisConfig}’ has no member named ‘CollectShapeRangeInfo’
         config.CollectShapeRangeInfo("./trt_cls_shape.txt");
                ^
/home/cps/PaddleOCR/deploy/cpp_infer/src/ocr_cls.cpp:123:16: 错误：‘using Config = struct paddle::AnalysisConfig {aka struct paddle::AnalysisConfig}’ has no member named ‘EnableTunedTensorRtDynamicShape’
         config.EnableTunedTensorRtDynamicShape("./trt_cls_shape.txt", true);
                ^
/home/cps/PaddleOCR/deploy/cpp_infer/src/structure_table.cpp: 在成员函数‘void PaddleOCR::StructureTableRecognizer::LoadModel(const string&)’中:
/home/cps/PaddleOCR/deploy/cpp_infer/src/structure_table.cpp:137:16: 错误：‘using Config = struct paddle::AnalysisConfig {aka struct paddle::AnalysisConfig}’ has no member named ‘CollectShapeRangeInfo’
         config.CollectShapeRangeInfo("./trt_table_shape.txt");
                ^
/home/cps/PaddleOCR/deploy/cpp_infer/src/structure_table.cpp:139:16: 错误：‘using Config = struct paddle::AnalysisConfig {aka struct paddle::AnalysisConfig}’ has no member named ‘EnableTunedTensorRtDynamicShape’
         config.EnableTunedTensorRtDynamicShape("./trt_table_shape.txt", true);
                ^
/home/cps/PaddleOCR/deploy/cpp_infer/src/structure_layout.cpp: 在成员函数‘void PaddleOCR::StructureLayoutRecognizer::LoadModel(const string&)’中:
/home/cps/PaddleOCR/deploy/cpp_infer/src/structure_layout.cpp:124:16: 错误：‘using Config = struct paddle::AnalysisConfig {aka struct paddle::AnalysisConfig}’ has no member named ‘CollectShapeRangeInfo’
         config.CollectShapeRangeInfo("./trt_layout_shape.txt");
                ^
/home/cps/PaddleOCR/deploy/cpp_infer/src/structure_layout.cpp:126:16: 错误：‘using Config = struct paddle::AnalysisConfig {aka struct paddle::AnalysisConfig}’ has no member named ‘EnableTunedTensorRtDynamicShape’
         config.EnableTunedTensorRtDynamicShape("./trt_layout_shape.txt", true);
                ^
/home/cps/PaddleOCR/deploy/cpp_infer/src/ocr_det.cpp: 在成员函数‘void PaddleOCR::DBDetector::LoadModel(const string&)’中:
/home/cps/PaddleOCR/deploy/cpp_infer/src/ocr_det.cpp:37:16: 错误：‘using Config = struct paddle::AnalysisConfig {aka struct paddle::AnalysisConfig}’ has no member named ‘CollectShapeRangeInfo’
         config.CollectShapeRangeInfo("./trt_det_shape.txt");
                ^
/home/cps/PaddleOCR/deploy/cpp_infer/src/ocr_det.cpp:39:16: 错误：‘using Config = struct paddle::AnalysisConfig {aka struct paddle::AnalysisConfig}’ has no member named ‘EnableTunedTensorRtDynamicShape’
         config.EnableTunedTensorRtDynamicShape("./trt_det_shape.txt", true);
                ^
make[2]: *** [CMakeFiles/ppocr.dir/src/ocr_rec.cpp.o] 错误 1
make[2]: *** 正在等待未完成的任务....
make[2]: *** [CMakeFiles/ppocr.dir/src/ocr_cls.cpp.o] 错误 1
make[2]: *** [CMakeFiles/ppocr.dir/src/structure_layout.cpp.o] 错误 1
make[2]: *** [CMakeFiles/ppocr.dir/src/structure_table.cpp.o] 错误 1
make[2]: *** [CMakeFiles/ppocr.dir/src/ocr_det.cpp.o] 错误 1
make[1]: *** [CMakeFiles/ppocr.dir/all] 错误 2
make: *** [all] 错误 2

**Comment by tangjinhan** [comment]
你好，我这边也出现一样的问题，这个问题你解决了吗 @haishi11

**Comment by yywangfei** [comment]
paddleocr的版本太高了，可以自己重新编译paddleinference库，或者降低paddleocr的版本，我改成release/2.5分支就好了
